src/mlpack/methods/reinforcement_learning/environment/cart_pole.hpp

```diff
@@ -0,0 +1,182 @@
+/**
+ * @file cart_pole.hpp
+ * @author Shangtong Zhang
+ *
+ * This file is an implementation of Cart Pole task
+ * https://gym.openai.com/envs/CartPole-v0
+ *
+ * TODO: refactor to OpenAI interface
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+
+#ifndef MLPACK_METHODS_RL_ENVIRONMENT_CART_POLE_HPP
+#define MLPACK_METHODS_RL_ENVIRONMENT_CART_POLE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace rl {
+
+namespace cart_pole_details {
+// Some constants of Cart Pole task
+constexpr double gravity = 9.8;
+constexpr double massCart = 1.0;
+constexpr double massPole = 0.1;
+constexpr double totalMass = massCart + massPole;
+constexpr double length = 0.5;
+constexpr double poleMassLength = massPole * length;
+constexpr double forceMag = 10.0;
+constexpr double tau = 0.02;
+constexpr double thetaThresholdRadians = 12 * 2 * 3.1416 / 360;
+constexpr double xThreshold = 2.4;
+}
+
+/**
+ * Implementation of Cart Pole task
+ */
+class CartPole
+{
+ public:
+
+  /**
+   * Implementation of state of Cart Pole
+   * Each state is a tuple of (position, velocity, angle, angular velocity)
+   */
+  class State
+  {
+   public:
+    //! Construct a state instance
+    State() : data(4) { }
+
+    //! Construct a state instance from given data
+    State(arma::colvec data) : data(data) { }
+
+    //! Get position
+    double X() const
```

Comment:
Another idea is to return the complete state instead of providing one function for each parameter, because it's basically used internally. That way, we could provide the same function set (State) for each environment.